plugins/tcpinfo/tcpinfo.cc

```diff
@@ -368,6 +415,15 @@ TSPluginInit(int argc, const char *argv[])
     delete config;
     return;
   }
+  if (TSTextLogObjectRollingEnabledSet(config->log, config->rolling_enabled) != TS_SUCCESS) {
+    TSError("[tcpinfo] failed to enable log file rolling to: '%d'", config->rolling_enabled);
+    delete config;
```

Comment:
Since you are deleting in multiple places, consider using `std::unique_ptr` and then doing

```C
  TSContDataSet(cont, config.release());
```